src/components/core/ContextSelector/index.tsx

```diff
@@ -1,176 +1,146 @@
-        if (currentContext && !types.find(type => currentContext.type.id === type)) {
-            exchangeCurrentContext();
-        }
-    }, [currentContext, types]);
+        search(queryText);
+    }, [queryText]);
 
     const setCurrentContextAsync = async (context?: Context) => {
-        if (context) {
-            await contextManager.setCurrentContextAsync(context);
-            setQueryText('');
-            setRelatedContexts([]);
-            setShouldShowResultsDropdown(false);
-        }
+        if (context) await contextManager.setCurrentContextAsync(context);
     };
 
-    const onSearchChange = React.useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-        setQueryText(e.target.value);
-    }, []);
+    const onKeyUpCloseDropDown = React.useCallback((e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (e.keyCode !== 27) return; //ESC
 
-    const onOpen = React.useCallback(() => setShouldShowResultsDropdown(true), []);
-    const onClose = React.useCallback(() => {
-        setShouldShowResultsDropdown(false);
+        setIsOpen(false);
         setQueryText('');
     }, []);
 
-    const onKeyUp = React.useCallback((e: React.KeyboardEvent<HTMLInputElement>) => {
-        if (e.keyCode === 27) {
-            // ESC
-            onClose();
-        }
+    const dropdownControllerProps = React.useCallback((ref, isOpen, setIsOpen) => {
+        const selectedItem = React.useMemo(() => {
+            const mergedItems = mergeDropdownSectionItems(dropdownSections);
+            const selectedItem = mergedItems.find(option => option.isSelected === true);
+            return selectedItem;
+        }, [dropdownSections]);
+
+        const selectedValue = React.useMemo(() => {
+            if (isOpen) {
+                return queryText;
+            } else if (selectedItem) {
+                return selectedItem.title;
+            } else if (currentContext) {
+                return currentContext.title;
+            }
+            return '';
+        }, [isOpen, queryText, selectedItem, currentContext]);
+
+        const onChangeQueryText = React.useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+            setQueryText(e.target.value);
+        }, []);
+
+        const onClickDropDown = React.useCallback(() => {
+            !isOpen && setIsOpen(true);
+        }, []);
+
+        return (
+            <>
+                <SearchIcon color="#DADADA" />
+                <input
+                    type="text"
+                    value={selectedValue}
+                    onChange={onChangeQueryText}
+                    onClick={onClickDropDown}
+                    onKeyUp={onKeyUpCloseDropDown}
+                    placeholder={selectedValue !== '' ? selectedValue : 'Search contexts'}
+                    className={styles.searchInput}
+                    ref={inputRef}
+                />
+                {isQuerying && <Spinner inline />}
+            </>
+        );
     }, []);

```

Comment:
Sould have a dependency on isOpen, queryText, selectedItem and currentContext

Reply:
isOpen and selectedItem is  declared within the useDropDownController. Not sure how to get the hook to react on itself in that sense. 
As the isOpen and selectedItem get pulled off of the object at line 104. 
And i cant really push that up. 

Added queryText and currentContext as of now.